pkg/placement/placement-decision.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/kubestellar/kubestellar/api/edge/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+// syncPlacementDecision syncs a placement-decision object with what is resolved by the placement-resolver.
+func (c *Controller) syncPlacementDecision(key util.Key) error {
+	var unstructuredObj *unstructured.Unstructured
+	if !c.placementResolver.ResolutionExists(key.NamespacedName.Name) {
+		// if a resolution is not associated to the placement-decision's name
+		// then the placement has been deleted, and the placement-decision has
+		// been garbage collected. We can safely ignore this.
```

Comment:
It is not guaranteed that the PlacementDecision has already been GCed, only that it eventually will be.

Reply:
To be sure, you're commenting on the accuracy of the text, and not the correctness of the code, right?

Reply:
Yes.